include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -200,6 +209,44 @@ class alignment_result
         return data.alignment;
     }
     //!\}
+
+    //!\cond DEV
+    /*!\brief Returns the score matrix used to compute the alignment.
+     * \return The score matrix.
+     *
+     * \details
+     *
+     * This function is only used for debugging such that performance can be affected significantly when enabling
+     * seqan3::align_cfg::debug.
+     *
+     * \attention This function with fail the compilation, if the alignment was not requested in the alignment
+     * configuration.
+     */
+    constexpr auto const & score_matrix() const noexcept
+    {
+        static_assert(!std::is_same_v<decltype(data.score_debug_matrix), std::nullopt_t *>,
+                      "Trying to access the score matrix, although it was not requested in the alignment configuration.");
+        return data.score_debug_matrix;
+    }
+
+    /*!\brief Returns the trace matrix used to compute the alignment.
+     * \return The score matrix.
+     *
+     * \details
+     *
+     * This function is only used for debugging such that performance can be affected significantly when enabling
+     * seqan3::align_cfg::debug.
+     *
+     * \attention This function with fail the compilation, if the alignment was not requested in the alignment
```

Comment:
Shouldn't you add that also the _debug_ config is required?

Reply:
Well, probably yes. It is only visible to developers though.